### ISSUE TITLE
Unable to integrate with the latest vite-react-swc template.

### ISSUE THREAD

**Issue by shlroland** [lingui/js-lingui#1988]
**To Reproduce**
Steps to reproduce the behavior, possibly with minimal code sample, e.g:

[lingui-test.zip](https://github.com/user-attachments/files/16455056/lingui-test.zip)

**Expected behavior**
Normal Operation

**Additional context**

- jsLingui version ``lingui --version``
- Babel version ``npm list @babel/core``
- Macro support:
 - [x] I'm using SWC with `@lingui/swc-plugin`
 - [ ] I'm using Babel with `babel-macro-plugin`
 - [ ] I'm not using macro
- Your Babel config (e.g. `.babelrc`) or framework you use (Create React App, NextJs, Vite)

computer: M1 Apple
OS: Macos 14

**Comment by shlroland** [comment]
@thekip

**Comment by thekip** [comment]
Sorry, i could not debug code for you. Did you check that version of SWC core is compatible with version of lingui swc plugin? What is the error message?

**Comment by shlroland** [comment]
> Sorry, i could not debug code for you. Did you check that version of SWC core is compatible with version of lingui swc plugin? What is the error message?

<img width="1198" alt="image" src="https://github.com/user-attachments/assets/a63f1535-c0f2-45d0-9e62-4dd57b01dc97">

```json
// package.json

{
  "name": "lingui-test",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "tsc -b && vite build",
    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
    "preview": "vite preview"
  },
  "dependencies": {
    "@lingui/macro": "^4.11.2",
    "@lingui/react": "^4.11.2",
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  "devDependencies": {
    "@lingui/cli": "^4.11.2",
    "@lingui/swc-plugin": "^4.0.8",
    "@lingui/vite-plugin": "^4.11.2",
    "@swc/core": "^1.7.4",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "@typescript-eslint/eslint-plugin": "^7.15.0",
    "@typescript-eslint/parser": "^7.15.0",
    "@vitejs/plugin-react-swc": "^3.5.0",
    "eslint": "^8.57.0",
    "eslint-plugin-react-hooks": "^4.6.2",
    "eslint-plugin-react-refresh": "^0.4.7",
    "typescript": "^5.2.2",
    "vite": "^5.3.5"
  }
}


```

**Comment by thekip** [comment]
You have incompatibility between swc-core and lingui swc plugin. Please remove `^` from `@swc/core` and `@lingui/swc-plugin`. 

Also check [compatibility table](https://github.com/lingui/swc-plugin?tab=readme-ov-file#compatibility)

**Comment by thekip** [comment]
Also this explcitly stated in the vite-swc example readme https://github.com/lingui/js-lingui/tree/main/examples/vite-project-react-swc#example-project-using-vite--swc-compiler-with-linguijs-plugin

and specific version of `@swc/core` fixed used resolutions field, as well as `@lingui/swc-plugin` specified without a caret symbol.

**Comment by basaran** [comment]
I had the same problem, did a pnpm update and swc that came with the `@vitejs/plugin-react-swc` got updated to 1.7 behind the scenes and lingui started failing. I checked from the history that the previous version was 1.6.5.

I added the following key to package.json:

```
    "resolutions": {
        "@swc/core": "~1.6.5"
    },
```

cleared node_modules, and .swc and executed a clean install. Now it's working and swc/core is set to 1.6.13.


P.S I couldn't make heads or tails of the compatibility matrix. It's a bit late here.
